File: Class 2/Problem 121/buyandsell.js
var maxProfit = function (prices) {
  let buy = prices[0];
  prices[0] = 0;
  let profit = 0;

  console.log(prices);

  for (i = 1; i < prices.length; i++) {
    if (buy > prices[i]) {
      buy = prices[i];
      prices[i] = 0;
    } else {
      profit = Math.max(prices[i] - buy, profit);
    }
  }

  return profit;
};

File: Class 2/Problem 121/buyandsell.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        buy = prices[0]
        prices[0] = 0
        profit = 0
        
        
        for i in range(1, len(prices)):
           print(prices[i])
           if buy > prices[i]:
              buy = prices[i]
              prices[i] = 0
           else:  
              profit = max(prices[i]-buy, profit)
        return profit